app/backend/src/tests/test_email.py

```diff
@@ -57,23 +58,101 @@ def test_report_email_rendering():
     assert subject in html
 
 
+def test_host_request_email_rendering():
+    subject = random_hex(64)
+    name_host = random_hex(64)
+    name_guest = random_hex(64)
+    from_date = "2020-01-01"
+    to_date = "2020-01-05"
+    host_request_link = random_hex(64)
+
+    plain, html = _render_email(
+        subject,
+        "host_request",
+        template_args={
+            "name_host": name_host,
+            "name_guest": name_guest,
+            "from_date": from_date,
+            "to_date": to_date,
+            "host_request_link": host_request_link
+        }
+    )
+
+    assert name_host in plain
+    assert name_host in html
+    assert name_guest in plain
+    assert name_guest in html
+    assert from_date in plain
+    assert from_date in html
+    assert to_date in plain
+    assert to_date in html
+    assert host_request_link in plain
+    assert host_request_link in html
+    assert subject in html
+
+
+def test_friend_request_email_rendering():
+    subject = random_hex(64)
+    name_recipient = random_hex(64)
+    name_sender = random_hex(64)
+    friend_requests_link = random_hex(64)
+
+    plain, html = _render_email(
+        subject,
+        "friend_request",
+        template_args={
+            "name_recipient": name_recipient,
+            "name_sender": name_sender,
+            "friend_requests_link": friend_requests_link
+        }
+    )
+
+    assert name_recipient in plain
+    assert name_recipient in html
+    assert name_sender in plain
+    assert name_sender in html
+    assert friend_requests_link in plain
+    assert friend_requests_link in html
+    assert subject in html
+
+
+def test_message_received_email_rendering():
+    subject = random_hex(64)
+    name_recipient = random_hex(64)
+    messages_link = random_hex(64)
+
+    plain, html = _render_email(
+        subject,
+        "message_received",
+        template_args={
+            "name_recipient": name_recipient,
+            "messages_link": messages_link
+        }
+    )
+
+    assert name_recipient in plain
+    assert name_recipient in html
+    assert messages_link in plain
+    assert messages_link in html
+    assert subject in html
+
+
 def test_login_email(db):
     user, api_token = generate_user(db)
 
     message_id = random_hex(64)
 
-    with session_scope(db) as session:
-        login_token, expiry_text = new_login_token(session, user)
+    login_token, expiry_text = new_login_token(db(), user)
```

Comment:
Please use the `session_scope(db)`, otherwise you can't guarantee the session gets closed, and that'll stall the tests if the tests wait for all connections to close before quitting!

You also need to indent the stuff after this line as they need the `login_token` object that's otherwise destroyed by sqlalchemy.